### PR TITLE
Fix typo in Snowflake dialect 

### LINKED ISSUE #3812: bug - typo in keyword in SnowFlake dialect
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

There's a typo in dialect_snowflake.py in line 1935. Sequence("APPLY", "RAW", "ACCESS", "POLICY") instead of Sequence("APPLY", "ROW", "ACCESS", "POLICY")

### Expected Behaviour

SQL statements should be parsed without errors

### Observed Behaviour

```
raise RuntimeError(
RuntimeError: Grammar refers to 'RawKeywordSegment' which was not found in the snowflake dialect. Perhaps specify the keyword? https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#keywords
```

### How to reproduce

1. Write a file.sql:
```
CREATE ROLE IF NOT EXISTS NEW_ROLE;

GRANT APPLY MASKING POLICY
  ON ACCOUNT TO ROLE NEW_ROLE;

GRANT CREATE MASKING POLICY
  ON SCHEMA MY_DB.MY_SCHEMA TO ROLE NEW_ROLE;
```
2. Run command:
sqlfluff lint file.sql --dialect snowflake

### Dialect

snowflake

### Version

sqlfluff 1.3.0
Python 3.9.12

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Gal40n04ek**
### Brief summary of the change made
fixes #3812
P.S. sorry for the typo I made yesterday in my PR

### Are there any other side effects of this change that we should be aware of?
nothing

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -1932,7 +1932,7 @@ class AccessStatementSegment(BaseSegment):
             ),
         ),
         Sequence("APPLY", "MASKING", "POLICY"),
-        Sequence("APPLY", "RAW", "ACCESS", "POLICY"),
+        Sequence("APPLY", "ROW", "ACCESS", "POLICY"),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:1935]: Maybe add a test case for this (and ideally others)?

As penance! 😉 And mroe seriously to catch any future regressions.

Comment (Gal40n04ek) [reply src/sqlfluff/dialects/dialect_snowflake.py:1935]: Sorry for that! I'll add tests now

Comment (Gal40n04ek) [reply src/sqlfluff/dialects/dialect_snowflake.py:1935]: I added tests for all cases I added yesterday. I'm very sorry for the typo I've made. I immediately made a new PR to fix it when I discovered this problem.

**Review by tunetheweb — APPROVED** [review]

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_snowflake.py:1935]: All good - no need to apologise!

**Pull request merged**